src/akkudoktoreos/utils/datetimeutil.py

```diff
@@ -16,165 +16,128 @@
+    elif isinstance(date_input, datetime):
+        dt = pendulum.instance(date_input)
     elif isinstance(date_input, date):
-        # Convert date object to datetime object
-        if to_maxtime is None or to_maxtime:
-            dt_object = datetime.combine(date_input, time.max)
-        else:
-            dt_object = datetime.combine(date_input, time.max)
+        dt = pendulum.instance(
+            datetime.combine(date_input, datetime.max.time() if to_maxtime else datetime.min.time())
+        )
     elif isinstance(date_input, (int, float)):
-        # Convert timestamp to datetime object
-        dt_object = datetime.fromtimestamp(date_input, tz=timezone.utc)
+        dt = pendulum.from_timestamp(date_input, tz="UTC")
     elif isinstance(date_input, str):
-        # Convert string to datetime object
-        try:
-            # Try ISO format
-            dt_object = datetime.fromisoformat(date_input)
-        except ValueError as e:
-            formats = [
-                "%Y-%m-%d",  # Format: 2024-10-13
-                "%d/%m/%y",  # Format: 13/10/24
-                "%d/%m/%Y",  # Format: 13/10/2024
-                "%m-%d-%Y",  # Format: 10-13-2024
-                "%Y.%m.%d",  # Format: 2024.10.13
-                "%d %b %Y",  # Format: 13 Oct 2024
-                "%d %B %Y",  # Format: 13 October 2024
-                "%Y-%m-%d %H:%M:%S",  # Format: 2024-10-13 15:30:00
-                "%Y-%m-%d %H:%M:%S%z",  # Format with timezone: 2024-10-13 15:30:00+0000
-                "%Y-%m-%d %H:%M:%S%z:00",  # Format with timezone: 2024-10-13 15:30:00+0000
-                "%Y-%m-%dT%H:%M:%S.%f%z",  # Format with timezone: 2024-10-13T15:30:00.000+0000
-            ]
-
-            for fmt in formats:
-                try:
-                    dt_object = datetime.strptime(date_input, fmt)
-                    break
-                except ValueError as e:
-                    dt_object = None
-                    continue
-            if dt_object is None:
-                raise ValueError(f"Date string {date_input} does not match any known formats.")
-    elif date_input is None:
-        if to_maxtime is None or to_maxtime:
-            dt_object = datetime.combine(date.today(), time.max)
+        dt = None
+        formats = [
+            "YYYY-MM-DD",  # Format: 2024-10-13
+            "DD/MM/YY",  # Format: 13/10/24
+            "DD/MM/YYYY",  # Format: 13/10/2024
+            "MM-DD-YYYY",  # Format: 10-13-2024
+            "D.M.YYYY",  # Format: 1.7.2024
+            "YYYY.MM.DD",  # Format: 2024.10.13
+            "D MMM YYYY",  # Format: 13 Oct 2024
+            "D MMMM YYYY",  # Format: 13 October 2024
+            "YYYY-MM-DD HH:mm:ss",  # Format: 2024-10-13 15:30:00
+        ]
+        for fmt in formats:
+            try:
+                # We assume values without timezone are specified in local date and time.
+                dt = pendulum.from_format(date_input, fmt, tz=pendulum.local_timezone())
```

Comment:
the Github pipeline fails because it has UTC as local timezone, should probably take in_timezone here if set.

Reply:
This is now fixed. Tests are passing.